### ISSUE TITLE
Unable to debug, memory breakpoint exception!

### ISSUE THREAD

**Issue by x64bugreport** [x64dbg/x64dbg#1854]
- Debugger version Dec 26 2017,13:38:00
- Operating system version and Service Pack win10 1709 and win7 sp1
Debugged program：https://raw.githubusercontent.com/x64bugreport/PlaneWar/7fc6521a3513d6b8b394364df98a58a9efa0b5fb/game.exe
1.Direct debugging will appear blank interface
2.After the addition, the memory breakpoint is abnormal and can not be broken accurately. For example, after the breakpoint at 0x00406D80, the address near the breakpoint will be operated.

**Comment by mrexodia** [comment]
Memory breakpoints currently only work on sizes in multiples of 0x1000,
breakpoint on 0x406d80 will break between 0x406000 and 0x407000.

**Comment by strivexjun** [comment]
Hello, I've been in this kind of problem, too.
In theory, the memory breakpoints are exactly what you say.
But OllyDbg sets a memory breakpoint for an address range, and he can work correctly. I think he should judge whether the target operand is within the range.

**Comment by mrexodia** [comment]
Yes I’m aware how to implement this and actually have in GleeBug. For now I
added a variable “$breakpointexceptionaddress” that you can use to do the
address check yourself in a conditional breakpoint.

**Comment by x64bugreport** [comment]
I know the range of 0x1000, excuse me, why not designed to be abnormal when the memory attribute is interrupted, if it is a non-defined address memory breakpoint debugger to deal with the release, this does affect the efficiency, but it does have a role, ollydbg can achieve these functions , Look forward to x32dbg achieve, In addition, I directly debug the sample program running this example, why the interface is not normal, there is no interface with olldbg display problems, may I ask you to reproduce it?

**Comment by x64bugreport** [comment]
This code can not debug properly run, to determine not due to anti-debugging and other causes, if you can reproduce the problem, I hope to give to solve the other want the memory breakpoint function can achieve ollydbg so.

**Comment by shuheque** [comment]
3 years, any progress on this item?

**Comment by mrexodia** [comment]
Duplicate of https://github.com/x64dbg/x64dbg/issues/1513
